polaris/evaluate/_metric.py

```diff
@@ -116,23 +173,102 @@ def score(
         assert metric.score(y_true=first, y_pred=second) == metric(y_true=first, y_pred=second)
         ```
         """
-        if y_pred is None and y_prob is None:
-            raise ValueError("Neither `y_pred` nor `y_prob` is specified.")
-
-        if self.y_type == "y_pred":
-            if y_pred is None:
-                raise ValueError(f"{self} requires `y_pred` input. ")
-            pred = y_pred
-        else:
-            if y_prob is None:
-                raise ValueError(f"{self} requires `y_prob` input. ")
-            pred = y_prob
+        y_true_targets = y_true.as_array("y")
+        pred = check_predictions(y_pred, y_prob, self.y_type)
 
-        kwargs = {"y_true": y_true, self.y_type: pred}
+        # Because we don't have multi-task metrics yet, we can assume this is a single-task metric
+        # With single-task metrics for multi-task benchmarks, it can happen that some target values are NaN.
+        # We mask these here.
+        mask = mask_index(y_true_targets)
+        kwargs = {"y_true": y_true_targets[mask], self.y_type: pred.mask(mask).flatten()}
         return self.fn(**kwargs, **self.value.kwargs)
 
     def __call__(
-        self, y_true: np.ndarray, y_pred: np.ndarray | None = None, y_prob: np.ndarray | None = None
+        self,
+        y_true: GroundTruth,
+        y_pred: BenchmarkPredictions | None = None,
+        y_prob: BenchmarkPredictions | None = None,
+    ) -> float:
+        return self.score(y_true, y_pred, y_prob)
+
+
+class GroupedMetric(BaseModel):
+    """"""
+
+    metric: Metric
+
+    group_by: str
+    on_error: Literal["ignore", "raise", "default"] = "raise"
+    default: float | None = None
+    aggregation: Literal["mean", "median"] = "mean"
+
```

Comment:
**Suggestion**
Since this model is considered to be equivalent to the metric it wraps, for the intent of evaluating uniqueness, I'd suggest adding equal and hashcode implementations that will make that check easier:

```python
def __eq__(self, other: Self | Metric) -> bool:
    match other:
        case Metric():
            return other == self.metric
        case GroupedMetric():
            return self.metric == other.metric
        case _:
            return False

def __hash__(self) -> int:
    return hash(self.metric)
```